src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -136,36 +159,38 @@ public StorageHelper(Context ctx) {
      * @return
      * @throws NoSuchAlgorithmException
      */
-    final private void loadSecretKeyForAPI() throws NoSuchAlgorithmException {
+    public SecretKey loadSecretKeyForAPI(Context context) throws IOException, GeneralSecurityException {
         // Loading key only once for performance. If API is upgraded, it will
         // restart the device anyway. It will load the correct key for new API. 
         if (mKey != null && mMacKey != null)
```

Comment:
Do we need to worry about thread safety? It looks like code wasn't really handling it before.